test-apps/sediment/src/main/java/com/twosigma/waiter/sediment/Servlet.java

```diff
@@ -0,0 +1,222 @@
+
+        final String correlationId = retrieveCorrelationId(request);
+        LOGGER.info(correlationId + " received " + request.getProtocol() + " " + request.getMethod() +
+            " request at " + request.getRequestURI());
+        response.setHeader("x-cid", correlationId);
+        response.setHeader("server", "sediment:" + HttpConfiguration.SERVER_VERSION);
+
+        final Request jettyRequest = (Request) request;
+        final Response jettyResponse = (Response) response;
+
+        switch (request.getRequestURI()) {
+            case "/":
+            case "/status": {
+                handleStatusRequest(jettyRequest, jettyResponse);
+                break;
+            }
+            case "/trailers": {
+                handleTrailersRequest(jettyRequest, jettyResponse);
+                break;
+            }
+            default: {
+                LOGGER.severe(correlationId + " unsupported endpoint: " + request.getRequestURI());
+                super.service(request, response);
+                break;
+            }
+        }
+        LOGGER.info(correlationId + " done processing request");
+    }
+
+    void handleStatusRequest(final Request request, final Response response) throws IOException {
+        final String contentParam = request.getParameter("content");
+        final String responseContent = contentParam == null ? "OK" : contentParam;
+        response.setContentType("text/plain");
+        response.getWriter().write(responseContent);
+    }
+
+    void handleTrailersRequest(final Request request, final Response response) throws IOException {
+        final Map<String, Object> infoMap = populateRequestInfo(request);
+        final String responseBody = new Gson().toJson(infoMap);
+        final byte[] outputBytes = responseBody.getBytes();
+
+        final HttpFields trailer = retrieveTrailers(request);
+        if (trailer != null) {
+            response.setTrailers(() -> trailer);
+        }
+
+        final String responseStatusString = request.getHeader("x-sediment-response-status");
+        if (responseStatusString == null) {
+            response.setStatus(200);
+        } else {
+            response.setStatus(Integer.parseInt(responseStatusString));
+        }
+
+        final ByteArrayInputStream respInputStream = new ByteArrayInputStream(outputBytes);
+        final ServletOutputStream respOutputStream = response.getOutputStream();
+        while (true) {
+            final byte[] tempBytes = new byte[128];
+            final int bytesRead = respInputStream.read(tempBytes);
+            if (bytesRead == -1) {
+                final String sleepMsString = request.getHeader("x-sediment-sleep-before-response-trailer-ms");
+                if (sleepMsString != null) {
+                    try {
+                        LOGGER.info("sleeping " + sleepMsString + " ms before returning response trailers");
+                        Thread.sleep(Long.parseLong(sleepMsString));
```

Comment:
This sleep is potentially preventing the emission of the terminating chunk.  The goal is to introduce the delay between the terminating chunk and terminating CRLF.

Reply:
As we spoke offline, jetty doesn't allow us to do that. I have added a similar behavior to kitchen (and added a test), but it works only for http 1.1 servers.